src/components/VideoPlayer/BaseVideoPlayer.js

```diff
@@ -0,0 +1,180 @@
+
+    // update shared video elements
+    useEffect(() => {
+        if (shouldUseSharedVideoElement || url !== currentlyPlayingURL) {
+            return;
+        }
+        shareVideoPlayerElements(videoPlayerRef.current, videoPlayerElementParentRef.current, videoPlayerElementRef.current);
+    }, [currentlyPlayingURL, shouldUseSharedVideoElement, shareVideoPlayerElements, updateSharedElements, url]);
+
+    // append shared video element to new parent (used for example in attachment modal)
+    useEffect(() => {
+        if (url !== currentlyPlayingURL || !sharedElement || !shouldUseSharedVideoElement) {
+            return;
+        }
+
+        const newParentRef = sharedVideoPlayerParentRef.current;
+        videoPlayerRef.current = currentVideoPlayerRef.current;
+        if (currentlyPlayingURL === url) {
+            newParentRef.appendChild(sharedElement);
+            bindFunctions();
+        }
+        return () => {
+            if (!originalParent && !newParentRef.childNodes[0]) {
+                return;
+            }
+            originalParent.appendChild(sharedElement);
+        };
+    }, [bindFunctions, currentVideoPlayerRef, currentlyPlayingURL, isSmallScreenWidth, originalParent, sharedElement, shouldUseSharedVideoElement, url]);
+    return (
+        <>
+            <Hoverable>
+                {(isHovered) => (
+                    <View style={[styles.w100, styles.h100, style]}>
+                        {shouldUseSharedVideoElement ? (
+                            <>
+                                <View
+                                    ref={sharedVideoPlayerParentRef}
+                                    style={[styles.flex1]}
+                                />
+                                {/* We are adding transparent absolute View between appended video component and control buttons to enable
+                                    catching onMouse events from Attachment Carousel. Due to late appending React doesn't handle
+                                    element's events properly. */}
+                                <View style={[styles.w100, styles.h100, styles.pAbsolute]} />
+                            </>
+                        ) : (
+                            <View
+                                style={styles.flex1}
+                                ref={(el) => {
+                                    if (!el) {
+                                        return;
+                                    }
+                                    videoPlayerElementParentRef.current = el;
+                                    if (el.childNodes && el.childNodes[0]) {
+                                        videoPlayerElementRef.current = el.childNodes[0];
+                                    }
+                                }}
+                            >
+                                <View style={styles.flex1}>
+                                    <Video
```

Comment:
Is there a way to toggle playing/pausing the video when you click/tap anywhere on the video, and not just the play/pause button?

Reply:
That can be tricky, I would do it as a followup task.

Reply:
@francoisl WDYT about implementing it as a followup?